modules/core/src/smithy4s/schema/FieldFilter.scala

```diff
@@ -158,10 +163,45 @@ object FieldFilter {
 
   val SkipNonRequiredDefaultValues: FieldFilter = skipNonRequiredDefaultValues
 
-  private case object skipUnsetOptions extends FieldFilter.SkipNonRequired {
-    def compileNonRequired[S, A](field: Field[S, A]): Predicate[A] = { a =>
-      a != None
+  private object IsNoneVisitor extends SchemaVisitor.Default[Predicate] {
+    def default[A]: Predicate[A] = Function.const(false)
+
+    override def biject[A, B](
+        schema: Schema[A],
+        bijection: Bijection[A, B]
+    ): Predicate[B] = {
+      if (schema.hints.has(Nullable)) {
```

Comment:
I'd rather this check was performed on the `def option` case. For consistency with the rest of the implementation that verify whether a field is nullable or not (https://github.com/disneystreaming/smithy4s/blob/series/0.18/modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala#L1356-L1357)